src/bf_functions.py

```diff
@@ -344,231 +291,184 @@ def frequency2fft_bin(self, frequencies=all):
-
-	if max(fft_bins) > n_chans or min(fft_bins) < 0:
-            raise fbfException(1, 'fft_bins out of range 0 -> %d' %(n_chans-1), \
-                               'function %s, line no %s\n' %(__name__, inspect.currentframe().f_lineno), \
-                               self.syslogger)
-
+        frequencies = []
+        n_chans = self.get_param('n_chans')
+        if fft_bins == all:
+            fft_bins = range(n_chans)
+        if type(fft_bins) == int:
+            fft_bins = [fft_bins]
+        if max(fft_bins) > n_chans or min(fft_bins) < 0:
+            raise fbfException(1, 'fft_bins out of range 0 -> %d' %(n_chans-1),
+                               'function %s, line no %s\n' %(__name__, inspect.currentframe().f_lineno), self.syslogger)
         bandwidth = self.get_param('bandwidth')
-
-	for fft_bin in fft_bins:
-	    frequencies.append((float(fft_bin)/n_chans)*bandwidth)
-
-	return frequencies
+        for fft_bin in fft_bins:
+            frequencies.append((float(fft_bin)/n_chans)*bandwidth)
+        return frequencies
 
     def frequency2fpga_bf(self, frequencies=all, fft_bins=[], unique=False):
         """returns a list of dictionaries {fpga, beamformer_index} based on frequency. unique gives only unique values"""
         locations = []
-
         if unique != True and unique != False:
-            raise fbfException(1, 'unique must be True or False', \
-                               'function %s, line no %s\n' %(__name__, inspect.currentframe().f_lineno), \
-                               self.syslogger)
-
+            raise fbfException(1, 'unique must be True or False', 'function %s, line no %s\n' %
+                               (__name__, inspect.currentframe().f_lineno), self.syslogger)
         if len(fft_bins) == 0:
             fft_bins = self.frequency2fft_bin(frequencies)
-
-        fpgas = self.frequency2fpgas(fft_bins=fft_bins)
-        bfs = self.frequency2bf_label(fft_bins=fft_bins)
-
+            fpgas = self.frequency2fpgas(fft_bins=fft_bins)
+            bfs = self.frequency2bf_label(fft_bins=fft_bins)
         if len(fpgas) != len(bfs):
-            raise fbfException(1, 'fpga and bfs associated with frequencies not the same length', \
-                               'function %s, line no %s\n' %(__name__, inspect.currentframe().f_lineno), \
-                               self.syslogger)
+            raise fbfException(1, 'fpga and bfs associated with frequencies not the same length',
+                               'function %s, line no %s\n' %(__name__, inspect.currentframe().f_lineno), self.syslogger)
         else:
             pfpga = []
             pbf = []
-            for index in range(len(fpgas)):
-                fpga = fpgas[index]; bf = bfs[index]
-                if (unique == False) or (pfpga != fpga or pbf != bf):
-                    locations.append({'fpga': fpga, 'bf': bf})
-                pbf = bf
-                pfpga = fpga
-
+        for index in range(len(fpgas)):
```

Comment:
Again meaning is changed, block will now be run regardless of the if/else block above. Intended?